ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -189,52 +187,51 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-
             if (scroll) {
                 this.scrollToTop();
             }
-
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-
-                this.isPageBusy.pipe(
-                    tap(_ => this.isLoaded$.next(false)),
-                    debounceTime(this.timeToDebounce),
-                    filter(busy => !busy),
-                    take(1)
-                ).subscribe(_ => {
-                    this.isLoaded$.next(true);
-                    this.currentSectionIndex = nextIndex;
-                    this.visitedSectionIndexes[nextIndex] = true;
-                    this.saveLastVisitedSectionIndex(nextIndex);
-                });
-            }
+            this.submitAnnouncementService.announceSubmit(null);
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(() => {
+                            this.validationRequested = true;
+                            this.sendSectionAnalytics();
+                            this.currentSection.validate();
+                            if (this.currentSection.valid) {
+                                this.resetValidationState();
+                                this.visitedSectionIndexes[nextIndex] = true;
+                                this.saveLastVisitedSectionIndex(nextIndex);
+                                this.currentSectionIndex = nextIndex;
+                            }
+                        })
+                    )
+                ),
+                take(1)
+            ).subscribe();
         }
     }
 
     public decrementStep(scroll: boolean = true): void {
         const previousIndex = this.previousAvailableSectionIndex();
         if (previousIndex !== -1) {
-            // if we move forwards or backwards, let's reset our validation display
-            this.resetValidationState();
-
             if (scroll) {
                 this.scrollToTop();
             }
-
-            this.isPageBusy.pipe(
-                tap(_ => this.isLoaded$.next(false)),
-                debounceTime(this.timeToDebounce),
-                filter(busy => !busy),
+            this.isPageBusy.pipe(startWith(true)).pipe(
+                delay(1),
+                filter(pageIsBusy => !pageIsBusy),
+                debounceTime(250),
```

Comment:
let's use this.timeToDebounce

Reply:
f49d5e7f83d0349d6db5750a2f7642c2f11bf8b5